### PR TITLE
refactor: migrate custom hook pattern

### PR DESCRIPTION
**Pull request by LiKang6688**
Migrate from
```
const [filteredData, setFilteredData] = useState();
useEffect(() => {
  setFilteredData(data.filter(x => isOfSomeType(x)));
}, [data]);
return <List data={filteredData} />;
```

to it
```
const filteredData = useMemo(() => {
  return data.filter(x => isOfSomeType(x));
}, [data]);
return <List data={filteredData} />;
```
Reference: https://jnystad.no/post/not-using-usecallback-is-premature-optimization

### CODE REVIEW

File: src/nebula-hooks/use-extended-theme.ts [hunk 1]
```diff
@@ -44,12 +44,10 @@ export const tableThemeColors = (theme: ExtendedTheme, rootElement: HTMLElement)
 const useExtendedTheme = (rootElement: HTMLElement): ExtendedTheme => {
   // TODO: add name method to stardust.Theme
   const nebulaTheme = useTheme() as ExtendedTheme;
-  const theme = useMemo(
+  return useMemo(
```
Comment (haxxmaxx) [inline src/nebula-hooks/use-extended-theme.ts:47]: this works the same? never seen a useMemo been directly returned

Comment (LiKang6688) [reply src/nebula-hooks/use-extended-theme.ts:47]: yes, typescript can help to verify that the type is the same. I will look into how to write unit tests for the custom hook.

**Review by haxxmaxx — APPROVED** [review]

File: src/nebula-hooks/use-extended-theme.ts (hunk 1 above)
Comment (haxxmaxx) [reply src/nebula-hooks/use-extended-theme.ts:47]: yeah I mean if there is a simple way to do it. We have a really high test coverage so more tests does not necessarily add more value

Comment (LiKang6688) [reply src/nebula-hooks/use-extended-theme.ts:47]: Okay, it is not easy. then I will not do it.

**Pull request merged**